File: spring/testconfiguration-example/src/main/java/com/example/so69782972/ClockConfig.java
package com.example.so69782972;

import java.time.Clock;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

@Configuration
public class ClockConfig {
    @Bean
    public Clock clock() {
        return Clock.systemDefaultZone();
    }
}

File: spring/clock-bean/src/main/kotlin/com/example/clock_bean/ClockConfig.kt
package com.example.clock_bean

import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import java.time.Clock
import java.time.ZoneId

@Configuration(proxyBeanMethods = false)
class ClockConfig {

    @Bean
    fun clock(): Clock {
        return Clock.system(ZoneId.of("Asia/Tokyo"))
    }
}
